### PR TITLE
ACS-1583: Original PoC: Storage classes - Produce OpenAPI spec for API

### PR DESCRIPTION
**Pull request by aforascu**
- add storage classes in the api specs for the following endpoints:
GET /storage-classes
GET /nodes/{nodeId}
GET /nodes/{nodeId}/children
POST /nodes/{nodeId}/children
PUT /nodes/{nodeId}

### CODE REVIEW

**Review by DenisGabriela — APPROVED** [review]

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 1]
```diff
@@ -7521,6 +7538,44 @@ paths:
...
+      tags:
+        - storage-classes
+      summary: Retrieve the list of supported storage classes in the system
+      description: |
+        **Note:** this endpoint is available in Alfresco 7.1 and newer versions.
+
+        Gets a list of all available storage classes
+
+          The default sort order for the returned list of storage classes is by ascending id.
+          You can override the default by using the **orderBy** parameter.
```
Comment (cristianturlica) [inline src/main/webapp/definitions/alfresco-core.yaml:7553]: We can't order the list.

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 2]
```diff
@@ -7521,6 +7538,44 @@ paths:
...
+          You can use any of the following fields to order the results:
+          * id
+      operationId: listStorageClasses
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/skipCountParam'
+        - $ref: '#/parameters/maxItemsParam'
+        - $ref: '#/parameters/orderByParam'
+        - $ref: '#/parameters/fieldsParam'
```
Comment (cristianturlica) [inline src/main/webapp/definitions/alfresco-core.yaml:7564]: We do not have pagination.

**Comment by tzclucian** [comment]
@aforascu , I'm not sure whether I miss it or not, but we need to make sure we specify that for the GET /node endpoint we should have `includeStorageClasses` as optional.

**Comment by montgolfiere** [comment]
Closing without merge. The original PoC work is now superseded by new design  (wip).

Including:
- https://github.com/Alfresco/rest-api-explorer/pull/139
- https://github.com/Alfresco/rest-api-explorer/pull/138
- ...

**Pull request closed without merging**